src/main/java/com/library/java_library_management/repository/TestModeRepository.java

```diff
@@ -0,0 +1,110 @@
+package com.library.java_library_management.repository;
+
+import com.library.java_library_management.dto.BookInfo;
+import com.library.java_library_management.status.BookStatus;
+
+import java.time.LocalDateTime;
+import java.util.*;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+public class TestModeRepository implements Repository{
+    private static final ScheduledExecutorService scheduler = Executors.newScheduledThreadPool(1);
+    private HashMap<Integer, BookInfo> bookMap = new HashMap<>();
```

Comment:
```suggestion
    private final Map<Integer, BookInfo> bookMap = new ConcurrentHashMap<>();
```

필드 선언 시 인터페이스를 선언부에 놓고 특정 구현체로 초기화하는 것이 일반적인 컨벤션 입니다.

Reply:
네 감사합니다! 근데 ConcurrentHashMap은 HashMap에 비해 멀티쓰레드 환경에서의 이점이 있는걸로 이해하고 있는데 여기서 ConcurrentHashMap 사용 권장이 되는 이유를 알 수 있을까요? 동시성 문제가 발생할 수 있는 상황 때문인가요?

Reply:
ScheduledExecutorService에 대한 로직이 섞여있어 동시성 문제가 발생할 수도 있기 때문이라고 이해했습니다. 맞다면 답변 안 주셔도 괜찮을 것 같습니다!

Reply:
팀 면담때 이야기 했던 것처럼 현재 과제 내용에서는 크게 문제될 내용은 아니지만, 
공유 자원을 사용한다는 개념으로 접근 했을때 자바 프로그램에서 고려해야 하는 내용중에 하나이기 때문에 언급하였습니다.

Reply:
넵 이해했습니다.

Reply:
추가적으로 repository에선 데이터 관리만 수행하도록 하고 service에서 비즈니스 로직을 수행하도록 개선해보았습니다.